kymatio/scattering2d/core/scattering2d.py

```diff
@@ -10,11 +10,16 @@ def scattering2d(x, pad, unpad, backend, J, L, phi, psi, max_order,
     irfft = backend.irfft    
     cdgmm = backend.cdgmm
     concatenate = backend.concatenate
+    to_complex = backend.to_complex
+
 
     # Define lists for output.
     out_S_0, out_S_1, out_S_2 = [], [], []
-
-    U_r = pad(x)
+    
+    if not pre_pad:
+        U_r = pad(x)
+    else:
+        U_r = to_complex(x)
```

Comment:
Remove these. The only use seems to be in Torch where we need to add a singleton dimension at the end. This can be done in the frontend and have `pad` assume the input has that shape to begin with.